ui/components/Filters.js

```diff
@@ -224,9 +234,39 @@ function MesheryFilters({ updateProgress, enqueueSnackbar, closeSnackbar, user,
     CLONE_FILTERS : {
       name : "CLONE_FILTER",
       error_msg : "Failed to clone filter file"
+    },
+    PUBLISH_CATALOG : {
+      name : "PUBLISH_CATALOG",
+      error_msg : "Failed to publish catalog"
+    },
+    UNPUBLISH_CATALOG : {
+      name : "PUBLISH_CATALOG",
+      error_msg : "Failed to publish catalog"
     }
   };
 
+/**
+   * Checking whether users are signed in under a provider that doesn't have
+   * publish filter capability and setting the canPublishFilter state accordingly
+   */
+useEffect(() => {
+  dataFetch(
+    "/api/provider/capabilities",
+    {
+      method : "GET",
+      credentials : "include",
+    },
+    (result) => {
+      if (result) {
+        const capabilitiesRegistry = result;
+        const filtersCatalogueCapability = capabilitiesRegistry?.capabilities.filter((val) => val.feature === MesheryFiltersCatalog);
+        if (filtersCatalogueCapability?.length > 0) setCanPublishFilter(true);
+      }
+    },
+    (err) => console.error(err)
```

Comment:
can we do something more with the error .

Reply:
Yes, please, @Chadha93. A toaster?